### PR TITLE
[Mailer][DX][RFC] Rename mailer bridge transport classes

### PR DESCRIPTION
**Pull request by Koc**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yno
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

During working on #31946 I realized how painful to work with multiple classes which has same name. [Nice article](https://www.tomasvotruba.cz/blog/2019/05/02/alias-as-a-code-smell/) by @TomasVotruba with explanation of problems  with such approach.
~Built on top of #32608 , so only [2nd commit](https://github.com/symfony/symfony/commit/bbf7e99e89c70fab372929827ae509b41280ce40) is actual.~

Also I've changed namespaces to make bridge structure much simpler and be linear. All classes located on same level now. See how [bridge](https://github.com/symfony/symfony/tree/bbf7e99e89c70fab372929827ae509b41280ce40/src/Symfony/Component/Mailer/Bridge/Amazon) looks like now.

Now in RFC state to get approve for such king of changes and update all other bridges.

### CODE REVIEW

**Comment by fabpot** [comment]
The renaming looks like a good idea to me 👍

**Comment by nicolas-grekas** [comment]
Please update the CHANGELOG file.

**Comment by Koc** [comment]
Branch rebased, PR is ready to review.

1st commit about renaming classes in bridges
2nd about some cleanup of built transports: abstract http and api classes moved under transport, in same time transport implementations moved under corresponsing sub-namespaces. So now `Transport` sub-namespace [contains only abstractions + Dsn](https://github.com/symfony/symfony/tree/b9dd600ffd4a347148b57af359fdc2df7cdf576d/src/Symfony/Component/Mailer/Transport)

File: src/Symfony/Component/Mailer/Bridge/Amazon/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,15 @@
 CHANGELOG
 =========
 
+4.4.0
+-----
+
+ * [BC BREAK] Renamed and moved `Symfony\Component\Mailer\Bridge\Amazon\Http\Api\SesTransport`
```
[the remark this answers is not in the document]
Comment (fabpot) [reply src/Symfony/Component/Mailer/Bridge/Amazon/CHANGELOG.md]: If you want, but I don't think that having FQCN is an issue.

[the remark this answers is not in the document]
Comment (Koc) [reply src/Symfony/Component/Mailer/Bridge/Amazon/CHANGELOG.md]: ok, let's leave FQCN

**Comment by fabpot** [comment]
Having shorter namespaces is also a goal. I would revert the creation of the `Chain`, `Null`, and `Sendmail` namespaces.

**Comment by Koc** [comment]
But in this case we got mess inside `Transport` subnamespace: a lot of classes, abstractions mixed with implementations.

**Comment by Koc** [comment]
изображение
vs
изображение

**Comment by fabpot** [comment]
But File and InMemory transports do not exist (and will probably never exist).

**Comment by Koc** [comment]
There are pending PRs about that: #31947, #32409

**Comment by fabpot** [comment]
Haven't had time to have a look at them yet, but I don't think they will be merged.

**Comment by Koc** [comment]
Creation of the `Chain`, `Null`, and `Sendmail` namespaces has been reverted.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Koc.

**Pull request merged**